CenterForOpenScience/osf.io
# [Feature] Optimize search migration [#OSF-9037]

## Purpose
Optimize `migrate_search` and `rebuild_search`

## Changes

* Use SQL to construct ES updates / deletes

Before: took some indeterminately large amount of time (10+ hours)
After: `rebuild` takes ~15 minutes
```
[website.search_migration.migrate]  INFO: 53188 nodes migrated
[website.search_migration.migrate]  INFO: 1170014 files migrated
[website.search_migration.migrate]  INFO: 74833 users migrated
inv rebuild_search  119.07s user 4.71s system 13% cpu 15:11.86 total
```
And `migrate` takes ~23
```
[website.search_migration.migrate]  INFO: 53188 nodes migrated
[website.search_migration.migrate]  INFO: 434963 nodes marked deleted
[website.search_migration.migrate]  INFO: 1170014 files migrated
[website.search_migration.migrate]  INFO: 2577084 files marked deleted
[website.search_migration.migrate]  INFO: 74833 users migrated
[website.search_migration.migrate]  INFO: 27706 users marked deleted
inv migrate_search  221.63s user 6.73s system 16% cpu 23:01.41 total
```

## QA Notes
Things to check:
* Non-ASCII characters on 'Search' and 'Add Contributor' pages
* Things that should be indexed are
* Things that shouldn't be indexed aren't (e.g. nodes tagged with `qatest`)
* Links work on search page
* Data is present and accurate (retractions behave normally, etc)

## Side Effects
None expected

## Ticket
[[OSF-9037]](https://openscience.atlassian.net/browse/OSF-9037)
